cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java

```diff
@@ -116,24 +116,38 @@ public void getData(HttpRequest request, HttpResponder responder,
   }
 
   @POST
-  @Path("/previews/{preview-id}/tracers/")
+  @Path("/previews/{preview-id}/tracers")
   public void getTracersData(HttpRequest request, HttpResponder responder,
                              @PathParam("namespace-id") String namespaceId,
                              @PathParam("preview-id") String previewId) throws Exception {
     NamespaceId namespace = new NamespaceId(namespaceId);
     ApplicationId application = namespace.app(previewId);
-    Map<String, List<String>> previewRequest = parseBody(request,
-                                                         new TypeToken<Map<String, List<String>>>() { }.getType());
-    if (previewRequest == null) {
-      responder.sendString(HttpResponseStatus.BAD_REQUEST, "Tracer name is not provided.");
-      return;
-    }
-    Map<String, Map<String, List<JsonElement>>> result = new HashMap<>();
-    for (List<String> tracerNames : previewRequest.values()) {
-      for (String tracerName : tracerNames) {
-        result.put(tracerName, previewManager.getRunner(application).getData(tracerName));
+    try {
+      Map<String, List<String>> previewRequest = parseBody(request,
+                                                           new TypeToken<Map<String, List<String>>>() {
+                                                           }.getType());
+      if (previewRequest == null) {
+        responder.sendString(HttpResponseStatus.BAD_REQUEST, "The body is not provided.");
+        return;
       }
+      Map<String, Map<String, List<JsonElement>>> result = new HashMap<>();
+      for (List<String> tracerNames : previewRequest.values()) {
+        if (tracerNames == null || tracerNames.isEmpty()) {
+          throw new BadRequestException("The tracer names cannot be empty.");
+        }
+        for (String tracerName : tracerNames) {
+          result.put(tracerName, previewManager.getRunner(application).getData(tracerName));
+        }
+      }
+      responder.sendString(HttpResponseStatus.OK, GSON.toJson(result));
+    } catch (JsonSyntaxException e) {
```

Comment:
this try-catch should just be around decoding the request body to make it clear what can cause the exception.